### PR TITLE
Add `installer-diff` program

### PR DESCRIPTION
**Pull request by andrew-farries**
## Description

Add a temporary  program to help compare the output of rendering the helm chart vs the gitpod installer.

This will be removed again once the https://github.com/gitpod-io/gitpod/issues/9097 epic is complete.

## Related Issue(s)

Relates to #9097

## Release Notes

```release-note
NONE
```

## Documentation

n/a

### CODE REVIEW

File: dev/installer-diff/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# installer-diff
+
```
Comment (geropl) [inline dev/installer-diff/README.md:2]: Maybe adding a "THIS IS TREMPORARY" + 1-2 sentence explanation would be nice.

File: dev/installer-diff/go.mod [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b // indirect
+	k8s.io/klog/v2 v2.30.0 // indirect
+	k8s.io/kube-openapi v0.0.0-20211115234752-e816edb12b65 // indirect
+	k8s.io/utils v0.0.0-20211116205334-6203023598ed // indirect
+	sigs.k8s.io/json v0.0.0-20211020170558-c049b76a60c6 // indirect
+	sigs.k8s.io/structured-merge-diff/v4 v4.2.1 // indirect
+	sigs.k8s.io/yaml v1.2.0 // indirect
+)
+
+replace k8s.io/api => k8s.io/api v0.23.4 // leeway indirect from components/common-go:lib
```
[the remark this answers is not in the document]
Comment (andrew-farries) [reply dev/installer-diff/go.mod:53]: done in 620bb18, which seems to have changed some other `go.mod` files too.

**Comment by geropl** [comment]
@andrew-farries Could you remove all non-`installer-diff` related changes from this PR, please? That should also get rid of the conflict.

**Comment by andrew-farries** [comment]
Rebased the head commit on the branch to exclude all the non-`installer-diff` changes that `leeway link` caused.

**Pull request merged**